### PR TITLE
[docs] Add development setup notes for Census API

### PR DESCRIPTION
**Pull request by prathapsridharan**
This is one of a series of PRs to improve developer documentation for the Census repository as stated in #1109

### CODE REVIEW

File: api/python/cellxgene_census/README.md [hunk 1]
```diff
@@ -7,3 +7,63 @@ The `cellxgene_census` package provides an API to facilitate the use of the CZ C
...
 
 If you believe you have found a security issue, we would appreciate notification. Please send email to <security@chanzuckerberg.com>.
+
+## Development Environment Setup
+
+- Create a virtual environment using `venv` or `conda`
+- `cd` to the root of this repository
+- `pip install -e api/python/cellxgene_census`
+- To install dependencies needed to work on the [experimental](./src/cellxgene_census/experimental/) portion of the API:
+  `pip install -e api/python/cellxgene_census[experimental]`.
```
Comment (ebezzi) [inline api/python/cellxgene_census/README.md:17]:
```suggestion
  `pip install -e 'api/python/cellxgene_census[experimental]'`.  
```

We can then remove the next line.

File: api/python/cellxgene_census/README.md [hunk 2]
```diff
@@ -7,3 +7,63 @@ The `cellxgene_census` package provides an API to facilitate the use of the CZ C
...
+607639  microwell-seq           neuron          adrenal gland  adrenal gland            cell  normal  female
+607640  microwell-seq           neuron          adrenal gland  adrenal gland            cell  normal  female
+
+[607641 rows x 7 columns]
+
+```
+
+- Learn more about the Census API by going through the tutorials in the [notebooks](../notebooks/)
+
+## Release
```
Comment (ebezzi) [inline api/python/cellxgene_census/README.md:67]: Not sure this belongs here. While the development setup can be useful for people who want to try the Census on their machine, this is only relevant for committers.

File: api/python/cellxgene_census/tests/experimental/ml/test_pytorch.py [hunk 3]
```diff
@@ -12,19 +12,15 @@
 from somacore import AxisQuery
 from tiledbsoma import Experiment, _factory
 from tiledbsoma._collection import CollectionBase
-from torch.utils.data._utils.worker import WorkerInfo
 
 # conditionally import torch, as it will not be available in all test environments
-try:
-    from torch import Tensor, float32
+from torch import Tensor, float32
```
Comment (ebezzi) [inline api/python/cellxgene_census/tests/experimental/ml/test_pytorch.py:17]: TODO: remove this once we figure out the issue.

Comment (prathapsridharan) [reply api/python/cellxgene_census/tests/experimental/ml/test_pytorch.py]: Reset the commit hash to remove it

File: api/python/cellxgene_census/README.md (hunk 2 above)
Comment (prathapsridharan) [reply api/python/cellxgene_census/README.md]: Removed

File: api/python/cellxgene_census/README.md (hunk 1 above)
Comment (prathapsridharan) [reply api/python/cellxgene_census/README.md]: Incorporated this suggestion

**Review by ebezzi — APPROVED** [review]

**Pull request merged**